scripts/update-readme.ts

```diff
@@ -0,0 +1,93 @@
+import { promises as fs } from "fs";
+import path from "path";
+
+interface TechMapping {
+  [key: string]: string;
+}
+
+interface Example {
+  tech: string;
+  name: string;
+  url: string;
+}
+
+const repoUrl =
+  "https://github.com/cloudinary-community/cloudinary-examples/tree/main/examples/";
+
+async function loadTechs(): Promise<TechMapping> {
+  const techsData = await fs.readFile("techs.json", "utf-8");
+  return JSON.parse(techsData);
+}
+
+async function getExamples(
+  techs: TechMapping
+): Promise<Record<string, Example[]>> {
+  const examplesByTech: Record<string, Example[]> = {};
+
+  const folders = await fs.readdir("examples");
+  for (const folder of folders) {
+    const configPath = path.join("examples", folder, "config.json");
+    try {
+      const configData = await fs.readFile(configPath, "utf-8");
+      const config = JSON.parse(configData);
+
+      // If tech exists in mapping, use mapped name, otherwise use "Other"
+      const tech = techs[config.tech] || techs["other"];
```

Comment:
do you think it would make sense to default to the lowercase version of the tech if it doesnt existi n the mapping?

say someone adds "angular" as the tech, could just list out "angular" or a title cased version

another idea is to throw an error if the tech doesn't exist to force it to create a match. on top of that, you can add a validation on Pull Request to make sure the tech exists in the list, if it doesnt, the person making the PR needs to add it to the list

Reply:
Yeah this makes sense, added these features in the latest commit